Class assignment-day 1/README.md

```diff
@@ -0,0 +1,58 @@
+# Day 1 Assignment
+
+- garbage collector
+- difference between arrow func and statement func. (binding)
+- hoisting
+- tdg (temporal dead zone)
+- script tag attributes
+- script tag async and defer
+- reduce function
+
+### Garbage Collection
+
+garbage collector is used to monitor memory allocation and determine when a block of allocated memory is no longer needed and reclaim it for further use. Garbage collection works by using references. If a object doesn't have any other object refering to it, then is is considered as garbage.
+
+let x = {
+
+    a: {
+        b: [1,2,3,4]
+    }
+
+};
+
+x = 4
+
+since, a doesn't have any variable referencing it, then it may be considered as garbage.
+
+### Difference between arrow function and statement function.
+
+The main difference between the arrow function and statement function in accordance with binding is: with statement functions the this keyword is bound to different values based on the context in which it is called.
```

Comment:
Can you also add some examples where you tried all this. Examples demonstrating how 'this' value changes in normal functions & arrow functions. Also check if call, apply & bind functions work on both